### DISCUSSION TITLE
Image filler

### DISCUSSION THREAD

**Discussion by Selyiah** [community/community#79151] [Programming Help] [answered]
### Body

Hiya! 

I'm not able to fill the image inside the card, I've tried to use object-fill: contain; however, I see no changes. I've also inspected the code on the browser and I can't seem to identify, which area is not triggering the correct response in order for the image to fill the card. Please advise, thanks community! 

<img width="1288" alt="Screenshot 2023-12-14 at 20 38 46" src="https://github.com/community/community/assets/140277594/2bbda2b6-7c45-4cf3-8ec9-563b80fed49e">

the HTML code: 

```
<h2>Hotspots of the week!</h2>

 <main class="main-area">
        
  <div class="centered">

      <section class="cards">
          
          <article class="card">
              <a href="#">
                  <picture class="thumbnail">
                      <img src="aged.jpg" alt="">
                  </picture>
                  <div class="card-content">
                  </div> <!-- .card-content -->
              </a>
          </article> <!-- .card -->

          <article class="card">
              <a href="#">
                  <picture class="thumbnail">
                      <img src="orange.jpg" alt="">
                  </picture>
                  <div class="card-content">
                  </div>  <!-- .card-content -->
              </a>
          </article> <!-- .card -->

          <article class="card">
              <a href="#">
                  <picture class="thumbnail">
                      <img src="prima.jpg" alt="">
                  </picture>
                  <div class="card-content">
                  </div> <!-- .card-content -->
             </a>
          </article> <!-- card --> 
```

the CSS code: 

```
h2 {
    padding: 15px;
}


/* the abbr and .card are to do with the background */
abbr,
acronym {
	border-bottom: 1px dotted #666;
	cursor: help;
}

.card {
	background: white;
	margin-bottom: 2em;	
} 

/* these two above are body */

.card a {
	color: black;
	text-decoration: none;
}

.card a:hover { 
	box-shadow: 3px 3px 8px hsl(0, 0%, 80%);    /* click and it takes you back to the top */
}

.card-content h2 {
	margin-top: 0;
	margin-bottom: .5em;
	font-weight: bold;
} 

/* Flexbox stuff */

.cards {
   display: flex;
   flex-direction: row;
   justify-content: space-evenly;
}
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ooopx** [comment]
Did you mean `object-fit: cover;` ? because using  `object-fit: contain;` will ![scale down your object to fit the set size](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit#examples) 
  
Also if you're using bootstrap, the scss from there can be adding flex or other properties to `.card` which will mess with the css you set. In this case you could try either using a different class name, or going back to inspect element to check if any scss is overriding your `.card` css definition

**Reply by Selyiah** [reply]
Got you, very helpful thank you! 

Do you think I could use `.thumbnail {` instead? or is it better to try to add `object-fit` in the `.card { `class ? 

I will inspect to see if there are some overriding components though, thank you

**Comment by varunpats** [comment]
Hey, try using `object-fit: cover;` that will be enough. If you are not happy after that, then try cropping image as per your liking and apply `object-fit: cover;`

**Reply by Selyiah** [reply]
Thank you for your help!

**Reply by varunpats** [reply]
No problem. If this helps could you please click `Mark as Answer` below, this will help me. Thanks!

**Comment by Selyiah** [comment]
I have tried but it still isn't working! Please see again community, thank you 

<img width="1342" alt="Screenshot 2023-12-19 at 21 09 14" src="https://github.com/community/community/assets/140277594/43b0c294-8a03-48cd-904c-0bbc4d1f8775">


I have tried adding `object-fit: cover;` and `background-size: cover;` and `transform: scale(1.2);` which the transform one did increase the image size but outside the card box. How can I get the image size to fit inside the card box? 

HTML code: 

```
<main class="main-area">
        
  <div class="card-deck">

      <section class="cards">
          
          <article class="card-1">
              <a href="#">
                <div class="card-image">
                    <img class="thumbnail" src="aged.jpg" alt="">
                </div>
              </a>
          </article> <!-- .card -->

          <article class="card-1">
              <a href="#">
                <div class="card-image">
                      <img class="thumbnail" src="orange.jpg" alt="">
                </div>
              </a>
          </article> <!-- .card -->

          <article class="card-1">
              <a href="#">
                <div class="card-image">
                      <img class="thumbnail-1" src="prima.jpg" alt="">
                </div>
             </a>
          </article> <!-- card --> 
        </div>
    </section>
    </main>

```
CSS code: 

```
abbr,
acronym {
	border-bottom: 1px dotted #666; 
	cursor: help;
}

.card-1 {
	text-decoration: none;
	border: 1px solid #808080;
	border-radius: 5px;
} 

/* these two above are body */

.card-1:hover { 
	box-shadow: 3px 3px 8px hsl(0, 0%, 80%);    /* click and it takes you back to the top */
}

.card-image {
	background-size: cover;
	width: 350px;
	height: 200px;
	background-repeat: no-repeat;
	background-position: center center;
}

/* Flexbox stuff */

.cards {
   display: flex;
   flex-direction: row;
   justify-content: space-evenly;
   background: white;
   margin-bottom: 2em;	
}
```

**Reply by ananuness** [reply]
i'm sorry, but you can't use background size, repeat, position etc in a div that doesn't have a background image! And setting fixed witdh and height can distort your image and not respect its aspect ratio, but if you want to set i would recommend to set directly in the img tag to work! You can check my comment too, to test if it works in your project!!

**Reply by Selyiah** [reply]
Thank you! Will check this

**Comment by davea42** [comment]
Nothing in css refers to class thumbnail or thumbnail-1
so what is the class name supposed to do?

**Reply by Selyiah** [reply]
I was referring should I make changes under this class name or not. I haven't put it in CSS yet because I wasn't sure

**Comment by ananuness** [comment, marked as answer]
Hi!! You can apply `object-fit: cover` (or contain) and `width: 100%` in the img tag. This way you will ensure that the image will stretch and adapt well in the container!

![image](https://github.com/community/community/assets/51466624/0175db7f-d422-4bbe-8313-b47157b89268)
> The *.thumbnail img* refers to the first html structure you provided! ☺️

If this doesn't work, maybe it is some other styles, then, if you're comfortable, you can share your entire css of this `cards`/`card-deck` container

I hope it helps 💕

**Reply by Selyiah** [reply]
Thank you so much! Will try this now

**Reply by Selyiah** [reply]
Thank you so much, it worked! Appreciate you

**Reply by Selyiah** [reply]
This is my CSS for it now since making the changes: 

```
abbr,
acronym {
	border-bottom: 1px dotted #666; 
	cursor: help;
}

.card-1 {
	text-decoration: none;
	border: 1px solid #808080;
	border-radius: 5px;
} 

/* these two above are body */

.card-1:hover { 
	box-shadow: 3px 3px 8px hsl(0, 0%, 80%);    /* click and it takes you back to the top */
}

.card-image {
	width: 350px;
	height: 230px;
}

.card-image-1 {
	width: 370px;
	height: 230px;
}

.thumbnail img {
	object-fit: cover;
	width: 100%;
	border-radius: 5px;
}

/* Flexbox stuff */

.cards {
   display: flex;
   flex-direction: row;
   justify-content: space-evenly;
   background: white;
   margin-bottom: 2em;	
}
```

& HTML since making the changes as well 

```
 <h2>Hotspots of the week!</h2>

 <main class="main-area">
        
  <div class="card-deck">

      <section class="cards">
          
          <article class="card-1">
              <a href="#">
                <div class="card-image">
                <picture class="thumbnail">
                    <img src="aged.jpg" alt="">
                </picture>
                </div>
              </a>
          </article> <!-- .card -->

          <article class="card-1">
              <a href="#">
                <div class="card-image">
                <picture class="thumbnail">
                    <img src="orange.jpg" alt="">
                </picture>
                </div>
              </a>
          </article> <!-- .card -->

          <article class="card-1">
              <a href="#">
                <div class="card-image-1">
                <picture class="thumbnail">
                    <img src="prima.jpg" alt="">
                </picture>
                </div>
             </a>
          </article> <!-- card --> 
        </div>
    </section>
    </main>
```

**Reply by ananuness** [reply]
Perfect!! It's well organized and I'm glad it worked 🤩
